library/core/src/option.rs

```diff
@@ -690,6 +690,74 @@ impl<T> Option<T> {
         }
     }
 
+    /////////////////////////////////////////////////////////////////////////
+    // Extracting slices
+    /////////////////////////////////////////////////////////////////////////
+
+    /// Extracts a slice that's empty if the option is a `None` value or
+    /// length one if the option is a `Some` value.
+    ///
+    /// Note that the slice extracted from a None value does not necessarily
+    /// contain an internal pointer to anything associated with the option.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// #![feature(option_as_slice)]
+    ///
+    /// let x: Option<u8> = Some(7);
+    /// assert_eq!(x.as_slice(), [7]);
+    ///
+    /// let x: Option<u8> = None;
+    /// assert_eq!(x.as_slice(), []);
+    /// ```
+    #[inline]
+    #[unstable(feature = "option_as_slice", issue = "none")]
+    pub fn as_slice(&self) -> &[T] {
```

Comment:
Can we make this `const`?

Reply:
Done in ac9bd888be8644315c0c0bcdec421dcf67f2373e.